### DISCUSSION TITLE
How to do TSX in Nuxt 3?

### DISCUSSION THREAD

**Discussion by fabis94** [nuxt/nuxt#17481] [Questions]
.tsx files seem to work automatically, but I can't figure out how to render custom components with JSX. If I do `<NuxtLink/>` with TSX, Nuxt doesn't seem to be able to resolve the component.

<img width="526" alt="image" src="https://user-images.githubusercontent.com/938316/167694480-f31c8de6-250b-4986-aba1-62705800de46.png">

<img width="474" alt="image" src="https://user-images.githubusercontent.com/938316/167694509-fda92450-89b2-4376-9664-5437e56ac45d.png">

**Comment by Mythie** [comment]
You can import these components by doing the following:

```typescript
import { NuxtLink } from '#components'
```

Your icon component can similarly be used by doing the following:

```typescript
const IconComponent = this.iconComponent;

return (<IconComponent />)
```

That said this doesn't seem to work with components like `<NuxtPage />` which makes this a non-starter for those of us that prefer TSX with Vue

**Reply by WuChenDi** [reply]
Thank you for your response. 

Do you have any additional documentation regarding `nuxt+tsx` development? I have found very little on the official website.

I have encountered a significant issue, namely, the automatic component importing function appears to be malfunctioning, although it is functioning correctly in `xxx.vue` files.
